tools/jvm/src/main/scala/org/scalajs/core/tools/linker/backend/closure/ClosureLinkerBackend.scala

```diff
@@ -54,6 +54,11 @@ final class ClosureLinkerBackend(
 
   val symbolRequirements: SymbolRequirement = emitter.symbolRequirements
 
+  private val needsIIFEWrapper = moduleKind match {
+    case ModuleKind.NoModule       => true
+    case ModuleKind.CommonJSModule => false
+  }
```

Comment:
Ah! Silly me. Of course because `ClosureLinkerBackend` always uses `ECMAScript51Isolated`.